Rock.Messaging.RabbitMQ/RabbitEngine.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Threading.Tasks;
+using RabbitMQ.Client;
+using RabbitMQ.Client.Events;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Rock.Messaging.RabbitMQ
+{
+    public class RabbitEngine : IReceiver, ISender
```

Comment:
Made this split, it's a lot cleaner. I base-classed the Connection creation. Currently installing Rabbit on a new machine to assess how much of the config I can pare back.